test/date.js

```diff
@@ -304,6 +316,116 @@ describe('date', () => {
+                });
+                Joi.date().timestamp('javascript').validate(milliseconds, (err, value) => {
+
+                    expect(err).to.not.exist();
+                    expect(value).to.deep.equal(now);
+                });
+                Joi.date().timestamp('unix').timestamp('javascript').validate(milliseconds, (err, value) => {
+
+                    expect(err).to.not.exist();
+                    expect(value).to.deep.equal(now);
+                });
+                done();
+            });
+
+            it('validates unix timestamp', (done) => {
+
+                const now = new Date();
+                const seconds = now.getTime() / 1000;
+
+                Joi.date().timestamp('unix').validate(seconds, (err, value) => {
+
+                    expect(err).to.not.exist();
+                    expect(value).to.deep.equal(now);
+                });
+                Joi.date().timestamp().timestamp('unix').validate(seconds, (err, value) => {
+
+                    expect(err).to.not.exist();
+                    expect(value).to.deep.equal(now);
+                });
+                Joi.date().timestamp('javascript').timestamp('unix').validate(seconds, (err, value) => {
+
+                    expect(err).to.not.exist();
+                    expect(value).to.deep.equal(now);
+                });
+                done();
+            });
+
+            it('validates timestamps with decimals', (done) => {
+
+                Helper.validate(Joi.date().timestamp(), [
+                    [new Date().getTime().toFixed(4), true]
+                ]);
+                Helper.validate(Joi.date().timestamp('javascript'), [
+                    [new Date().getTime().toFixed(4), true]
+                ]);
+                Helper.validate(Joi.date().timestamp('unix'), [
+                    [(new Date().getTime() / 1000).toFixed(4), true]
+                ]);
+                done();
+            });
+
+            it('validates only valid timestamps', (done) => {
+
+                Helper.validate(Joi.date().timestamp(), [
+                    [new Date().getTime(), true],
+                    [new Date().getTime().toFixed(4), true],
+                    ['1.452126061677e+12', true],
+                    [1.452126061677e+12, true],
+                    [1E3, true],
+                    ['1E3', true],
+                    [',', false],
+                    ['123A,0xA', false],
+                    ['1-1-2013 UTC', false],
+                    ['not a valid timestamp', false],
+                    [new Date('not a valid date'), false]
+                ], done);
+            });
+
+            it('fails with not allowed type', (done) => {
+
+                expect(() => {
+
+                    Joi.date().timestamp('not allowed');
+                }).to.throw(Error, /"type" must be one of/);
+                done();
+            });
+
+            it('validates timestamps with a friendly error message', (done) => {
```

Comment:
I really feel like this test should be merged with the 2 first functions of timestamp tests.